### PR TITLE
AB#8213 Promo codes applied before login are not transferred to the known user cart

### PR DESCRIPTION
**Pull request by chpa7**

### CODE REVIEW

File: Website/src/rendering/src/redux/ocCurrentCart/index.ts [hunk 1]
```diff
@@ -233,8 +238,29 @@ export const retrieveCart = createOcAsyncThunk<RequiredDeep<DOrderWorksheet> | u
       ) {
         ThunkAPI.dispatch(retrievePayments(existingOrder.ID));
       }
-      ThunkAPI.dispatch(retrievePromotions(existingOrder.ID));
+      const existingOrderPromos = (await ThunkAPI.dispatch(retrievePromotions(existingOrder.ID)))
```
Comment (jflheureux) [inline Website/src/rendering/src/redux/ocCurrentCart/index.ts:241]: The promos are not transferred with this scenario:
1. Log in
2. Complete an order until the Thank you page. (so you do not have any new empty order)
3. Log out
4. As an anonymous shopper, add products to cart, add promos, and go to the pre-checkout page.
5. From the pre-checkout page, log in.
Promos are not merged.

Comment (jflheureux) [inline Website/src/rendering/src/redux/ocCurrentCart/index.ts:241]: Have you tried to put this code in the `mergeAnonOrder` function? It seems that its job is to merge the orders, thus it seems to be the logical place to put this new code.

**Review by jflheureux — CHANGES_REQUESTED** [review]

File: Website/src/rendering/src/redux/ocCurrentCart/index.ts (hunk 1 above)
Comment (chpa7) [reply Website/src/rendering/src/redux/ocCurrentCart/index.ts]: Should be fixed now

**Review by jflheureux — APPROVED** [review]
I modified the ocCurrentCart logic. This is what I meant by merging the promos inside the mergeAnonOrder function. It works well for me locally. Maybe you can test it before merging.

**Comment by chpa7** [comment]
> I modified the ocCurrentCart logic. This is what I meant by merging the promos inside the mergeAnonOrder function. It works well for me locally. Maybe you can test it before merging.

Hmm I remember having a race condition sometimes, this is why I resorted to merging the promos long after the lineitems merge and fetching the worksheet once more. Anyway, it seems to be working now. Thanks!

**Pull request merged**